app/controllers/sessions_controller.rb

```diff
@@ -31,29 +32,62 @@ def login(json_response)
   def landing_page
     @user = current_user
   end
+
+  # profile page controller
+  # pass the user_type to landing_page.html.erb
+  # based on user_type serve a different view in the landing_page.html.erb
   def profile_page
+    @user = current_user
+    @user_type = session[:user_type]
   end
-  private
 
-  def log_out
-    session.delete(:user_id)
-    session.delete(:user_type)
-    @current_user = nil
+  def get_current_user
+    response = {
+        "user_type": "no user",
+        "user_id": -1
+    }
+    user_type = session[:user_type]
+    if user_type == "tenant"
+      response = {
+        "user_type": user_type,
+        "user_id": session[:user_id]
+      }
+    elsif user_type == "landowner"
+      response = {
+        "user_type": user_type,
+        "user_id": session[:user_id]
+      }
+    elsif user_type == "vendor"
+      response = {
+        "user_type": user_type,
+        "user_id": session[:user_id]
+      }
+    end
+    render json: response, status: :ok
   end
 
+  private
+
   def current_user
-    user_type = sessions[:user_type]
+    response = []
+    user_type = session[:user_type]
     if user_type == "tenant"
-      @current_user ||= Tenant.find_by(id: sessions[:user_id])
```

Comment:
not sure what the difference between sessions and session is but it didn't work until I changed it. Didn't seem to break anything either.

Reply:
calendar page also uses "session" instead of sessions